### PR TITLE
feat: split and inject fee curve parameters

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR:

- adds early support for splitting fee parameters into multiple models (swapper only for now, THORChain LP on the PR on top of this one)
- injects parameters instead of importing them from constants

This PR does not yet:

- implement THORChain LP fee curve

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- contributes to https://github.com/shapeshift/web/issues/6179

Low as this just moves things arounds but FOX discounts could be borked, test accordingly

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- FOX discounts are still happy

Note, you'll probably want to test this after a cache clear to ensure we're not leveraging cached FOX discounts and the testing is accurate

### Operations

- FOX discounts are still happy

## Screenshots (if applicable)

- FOX fully discounted swaps

image

- Undiscounted swaps

image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* **#6301** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6301?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/6301?utm_source=stack-comment).

File: src/lib/fees/parameters/types.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  FEE_CURVE_MIN_FEE_BPS: number
+  FEE_CURVE_NO_FEE_THRESHOLD_USD: number
+  FEE_CURVE_FOX_MAX_DISCOUNT_THRESHOLD: number
+  FEE_CURVE_MIDPOINT_USD: number
+  FEE_CURVE_STEEPNESS_K: number
+}
+
+export type ParameterModel = 'swapper'
+// TODO(gomes): Add THORChain LP model parameters and consume them in Thorchain LP domain
+// | 'thorchainLP'
```
Comment (gomesalexandre) [inline src/lib/fees/parameters/types.ts:12]: https://github.com/shapeshift/web/pull/6304

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx [hunk 2]
```diff
@@ -794,6 +795,8 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
     const { feeBps, feeUsd } = calculateFees({
       tradeAmountUsd: bn(totalAmountFiat),
       foxHeld: votingPower !== undefined ? bn(votingPower) : undefined,
+      // TODO(gomes)
+      parameters: feeCurveParameters.swapper,
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx:799]: https://github.com/shapeshift/web/pull/6304

**Review by kaladinlight — APPROVED** [review]
The only thing that crossed my mind is the constant style naming consistency. Wonder if we just keep the actual constants all caps, and then the parameters object looking like normal camel case keys. Could even `Object.freeze` the different parameters for some extra safety around it.

Alternative would be constant casing for the whole object. `FEE_CURVE_PARAMETERS.SWAPPER.SOME_PARAM`

Just feels weird with the mix to me at least, could be my ocd tendencies though lol

**Pull request merged**